Sources/SwiftParser/Lexer.swift

```diff
@@ -768,22 +774,51 @@ extension Lexer.Cursor {
   }
 }
 
+struct LexerResult {
```

Comment:
Can we make this an inner type `Lexer.Result`?

Reply:
Sure. I’m not sure how best to spell the return values though. Something like `return Result(.leftSquareBracket)` would be the most idiomatic in Swift but to me that looks like we’re returning the `Result` type from the stdlib. Thus I’ve gone with the fully-qualified `return Lexer.Result(.leftSquareBracket)` for now.

Reply:
Could name it `LexedToken` or something, not sure that's much better though 🤷

Reply:
I don’t think that’s an improvement.

Reply:
Yeah me neither :). I'm happy with what you changed it to.